demos/imagenet_classification_demo/main.cpp

```diff
@@ -0,0 +1,416 @@
+        std::vector<InferenceEngine::InferRequest> inferRequests;
+        int infReqNum = FLAGS_nireq;
+        for(int infReqID = 0; infReqID < infReqNum; ++infReqID)
+            inferRequests.push_back(executableNetwork.CreateInferRequest());
+
+        std::queue<cv::Mat> showMats;
+        
+        std::condition_variable condVar;
+        std::mutex mutex;
+        std::mutex showMutex;
+
+        //curImg is stored in ieWrapper
+        unsigned batchSize = FLAGS_b;
+        int irFirstIndex = 0;
+        
+        // --------------------------------------Set completion callback---------------------------------------
+        for (InferenceEngine::InferRequest& inferRequest : inferRequests) {
+            inferRequest.SetCompletionCallback(
+                InferRequestCallback(
+                    inferRequest,
+                    irFirstIndex,
+                    mutex,
+                    condVar,
+                    inputImgs,
+                    inputBlobName,
+                    FLAGS_b,
+                    showMats,
+                    curPos,
+                    framesNum,
+                    quitFlag
+                )
+            );
+            irFirstIndex = (irFirstIndex + batchSize) % inputImgs.size();
+        }
+        
+        // -------------------------------------------Filling blobs-------------------------------------------
+        auto blobDims = inputBlobsDimsInfo[inputBlobName];
+        if (blobDims.size() != 4) {
+            throw std::runtime_error("Input data does not match size of the blob");
+        }
+        for (InferenceEngine::InferRequest& inferRequest : inferRequests) { 
+            //setInputBlob(*it, curPos);    
+            auto inputBlob = inferRequest.GetBlob(inputBlobName);
+
+            for(unsigned i = 0; i < batchSize; i++) {        
+                cv::Mat inputImg = inputImgs.at((curPos+i)%inputImgs.size());
+                matU8ToBlob<uint8_t>(inputImg, inputBlob, i);
+            }
+            curPos = (curPos + batchSize) % inputImgs.size();   
+        }
+
+        // --------------------------------------Create output info---------------------------------------
+        int width;
+        int height;
+        std::vector<std::string> gmRowsCols = split(FLAGS_res, 'x');        
+        if (gmRowsCols.size() != 2) throw std::runtime_error("Input gmsizes key is not valid.");
+        else {
+            width = std::stoi(gmRowsCols[0]);
+            height = std::stoi(gmRowsCols[1]);
+        }
+        
+        GridMat gridMat = GridMat(cv::Size(width, height));
+
+        // ------------------------------------------Start async------------------------------------------
+        int64 startTime = cv::getTickCount();
+        for (InferenceEngine::InferRequest& inferRequest : inferRequests)
+            inferRequest.StartAsync();
+    
+    
+        int64 delay = FLAGS_delay;
```

Comment:
You can just use `FLAGS_*`. Why do you copy them?

Reply:
Done